cpp/src/strings/count_matches.cu

```diff
@@ -62,11 +63,14 @@ struct count_dispatch_fn {
 
   template <int stack_size>
   std::unique_ptr<column> operator()(column_device_view const& d_strings,
+                                     size_type output_size,
                                      rmm::cuda_stream_view stream,
                                      rmm::mr::device_memory_resource* mr)
   {
+    CUDF_EXPECTS(output_size >= d_strings.size(), "Unexpected output size");
```

Comment:
Since `count_matches` is a detail API used to implement these functions internally, this error would only ever be encountered if a developer made an error, right? So maybe an assertion is more appropriate? Just a minor suggestion though: the result of a smaller size would be an out of bounds access in the transform below, which I suppose may be hard for a developer to diagnose if it didn't fail in a reproducible way.

Reply:
Yes, I agree this is probably closer to an assert. I will look at changing it.